### PR TITLE
r/api_gateway_domain_name: Add support for Regional API Endpoints

### PR DESCRIPTION
**Pull request by xmackex**
Added support for Regional Endpoints in API GW

### CODE REVIEW

File: aws/resource_aws_api_gateway_domain_name.go [hunk 1]
```diff
@@ -52,6 +57,15 @@ func resourceAwsApiGatewayDomainName() *schema.Resource {
 				ConflictsWith: []string{"certificate_arn"},
 			},
 
+			"endpoint_configuration": {
+				Type: schema.TypeSet,
+				//Type:     schema.TypeSet,
+				//Optional: true,
```
Comment (Ninir) [inline aws/resource_aws_api_gateway_domain_name.go]: These 2 lines should be removed

File: aws/resource_aws_api_gateway_domain_name.go [hunk 2]
```diff
@@ -41,7 +41,12 @@ func resourceAwsApiGatewayDomainName() *schema.Resource {
 			"certificate_name": {
 				Type:          schema.TypeString,
 				Optional:      true,
-				ConflictsWith: []string{"certificate_arn"},
+				ConflictsWith: []string{"certificate_arn", "regional_certificate_arn", "regional_certificate_name"},
+			},
+			"regional_certificate_name": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"certificate_arn", "regional_certificate_name", "certificate_name"},
```
Comment (Ninir) [inline aws/resource_aws_api_gateway_domain_name.go]: Nitpick: don't you want to conflict with `regional_certificate_arn ` instead?

File: aws/resource_aws_api_gateway_domain_name.go [hunk 3]
```diff
@@ -61,7 +75,12 @@ func resourceAwsApiGatewayDomainName() *schema.Resource {
 			"certificate_arn": {
 				Type:          schema.TypeString,
 				Optional:      true,
-				ConflictsWith: []string{"certificate_body", "certificate_chain", "certificate_name", "certificate_private_key"},
+				ConflictsWith: []string{"certificate_body", "certificate_chain", "certificate_name", "certificate_private_key", "regional_certificate_arn"},
```
Comment (Ninir) [inline aws/resource_aws_api_gateway_domain_name.go]: Don't we want to conflict with `regional_certificate_name` also?

File: aws/resource_aws_api_gateway_domain_name.go [hunk 4]
```diff
@@ -61,7 +75,12 @@ func resourceAwsApiGatewayDomainName() *schema.Resource {
...
+			},
+			"regional_certificate_arn": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"certificate_body", "certificate_chain", "certificate_name", "certificate_private_key", "regional_certificate_name"},
```
Comment (Ninir) [inline aws/resource_aws_api_gateway_domain_name.go]: Don't we want to conflict with certificate_arn also?

**Comment by bflad** [comment]
This functionality was implemented in #2866 and will release in v1.20.0 of the AWS provider later this week.

**Pull request closed without merging**